code/datums/components/gun_auto_fire.dm

```diff
@@ -0,0 +1,119 @@
+
+
+/datum/component/automatedfire/gun
```

Comment:
I was asked to review this.
My opinion is that this part is a regression.
The bucket SS is a good concept, but as it is it only processes `/datum/component/automatedfire` types.
The autofire component can definitely be made more generic, but this makes it less so.
At this point it's no longer really a component.